packages/dd-trace/src/config.js

```diff
@@ -806,31 +544,285 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
+      process.env.DD_TRACE_AGENT_HOSTNAME,
+      '127.0.0.1'
+    )
+    return DD_AGENT_HOST || (url && url.hostname)
+  }
+
+  _getSpanComputePeerService () {
+    const DD_TRACE_SPAN_ATTRIBUTE_SCHEMA = validateNamingVersion(
+      coalesce(
+        this.options.spanAttributeSchema,
+        process.env.DD_TRACE_SPAN_ATTRIBUTE_SCHEMA
+      )
+    )
+
+    const peerServiceSet = (
+      this.options.hasOwnProperty('spanComputePeerService') ||
+      process.env.hasOwnProperty('DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED')
+    )
+    const peerServiceValue = coalesce(
+      this.options.spanComputePeerService,
+      process.env.DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED
+    )
+
+    const spanComputePeerService = (
+      DD_TRACE_SPAN_ATTRIBUTE_SCHEMA === 'v0'
+        // In v0, peer service is computed only if it is explicitly set to true
+        ? peerServiceSet && isTrue(peerServiceValue)
+        // In >v0, peer service is false only if it is explicitly set to false
+        : (peerServiceSet ? !isFalse(peerServiceValue) : true)
+    )
+
+    return spanComputePeerService
+  }
+
+  _isCiVisibilityGitUploadEnabled () {
+    return coalesce(
+      process.env.DD_CIVISIBILITY_GIT_UPLOAD_ENABLED,
+      true
+    )
+  }
+
+  _isCiVisibilityManualApiEnabled () {
+    return coalesce(
+      process.env.DD_CIVISIBILITY_MANUAL_API_ENABLED,
+      false
+    )
+  }
+
+  _isTraceStatsComputationEnabled () {
+    return coalesce(
+      this.options.stats,
+      process.env.DD_TRACE_STATS_COMPUTATION_ENABLED,
+      getIsGCPFunction() || getIsAzureFunctionConsumptionPlan()
+    )
+  }
+
+  _getTraceAgentUrl () {
+    return coalesce(
+      this.options.url,
+      process.env.DD_TRACE_AGENT_URL,
+      process.env.DD_TRACE_URL,
+      null
+    )
+  }
+
+  // handles values calculated from a mixture of options and env vars
+  _applyCalculated () {
+    const calc = this._calculated = {}
+
+    const {
+      DD_CIVISIBILITY_AGENTLESS_URL
+    } = process.env
+
+    if (DD_CIVISIBILITY_AGENTLESS_URL) {
+      this._setValue(calc, 'url', new URL(DD_CIVISIBILITY_AGENTLESS_URL))
+    } else {
+      this._setValue(calc, 'url', getAgentUrl(this._getTraceAgentUrl(), this.options))
+    }
+    if (this._isCiVisibility()) {
+      this._setBoolean(calc, 'isEarlyFlakeDetectionEnabled',
+        coalesce(process.env.DD_CIVISIBILITY_EARLY_FLAKE_DETECTION_ENABLED, true))
+    }
+    this._setString(calc, 'dogstatsd.hostname', this._getHostname())
+    this._setBoolean(calc, 'spanComputePeerService', this._getSpanComputePeerService())
+    this._setBoolean(calc, 'isIntelligentTestRunnerEnabled',
+      isTrue(this._isCiVisibility()) && isTrue(this._isCiVisibilityItrEnabled()))
+    this._setBoolean(calc, 'isGitUploadEnabled',
+      calc['isIntelligentTestRunnerEnabled'] && !isFalse(this._isCiVisibilityGitUploadEnabled()))
+    this._setBoolean(calc, 'isManualApiEnabled',
+      isTrue(this._isCiVisibility()) && isTrue(this._isCiVisibilityManualApiEnabled()))
```

Comment:
could we include these three inside `if (this._isCiVisibility())` ? I think it'd be way more readable

Reply:
why? they are not inside an `if` right now

Reply:
there's an `if` above that finishes but we're still using the same condition here: if we're going to use an if, why not use it to grab all the statements that we can? 
```javascript
if (this._isCiVisibility()) {
  this._setBoolean(calc, 'isEarlyFlakeDetectionEnabled',
    coalesce(process.env.DD_CIVISIBILITY_EARLY_FLAKE_DETECTION_ENABLED, true))
  this._setBoolean(calc, 'isIntelligentTestRunnerEnabled', isTrue(this._isCiVisibilityItrEnabled())) 
  this._setBoolean(calc, 'isManualApiEnabled', isTrue(this._isCiVisibilityManualApiEnabled()))
}
```

Reply:
oh i get what you mean now O-O this is so we can remove `isTrue(this._isCiVisibility()) &&`. make sense